front/components/assistant_builder/TeamSharingSection.tsx

```diff
@@ -9,74 +10,152 @@ import {
 } from "@dust-tt/sparkle";
 import type { AgentConfigurationScope, WorkspaceType } from "@dust-tt/types";
 import { isBuilder } from "@dust-tt/types";
+import { useState } from "react";
 
 import { assistantUsageMessage } from "@app/lib/assistant";
-import { useAgentUsage } from "@app/lib/swr";
+import { useAgentConfiguration, useAgentUsage } from "@app/lib/swr";
+
+type ConfirmationModalDataType = {
+  title: string;
+  text: string;
+  confirmText: string;
+  usageText?: string;
+  variant: "primary" | "primaryWarning";
+};
 
 /*
  * Note: Non-builders cannot change to/from company assistant
  */
 export function TeamSharingSection({
   owner,
   agentConfigurationId,
-  scope,
+  initialScope,
+  newScope,
   setNewScope,
 }: {
   owner: WorkspaceType;
   agentConfigurationId: string | null;
-  scope: Exclude<AgentConfigurationScope, "global">;
+  initialScope: Exclude<AgentConfigurationScope, "global">;
+  newScope: Exclude<AgentConfigurationScope, "global">;
   setNewScope: (scope: Exclude<AgentConfigurationScope, "global">) => void;
 }) {
+  const { agentConfiguration } = useAgentConfiguration({
+    workspaceId: owner.sId,
+    agentConfigurationId,
+  });
+  const assistantInMyList = agentConfiguration?.userListStatus === "in-list";
+  const assistantName = agentConfiguration?.name;
+
+  const [requestNewScope, setModalNewScope] = useState<Exclude<
+    AgentConfigurationScope,
+    "global"
+  > | null>(null);
   const agentUsage = useAgentUsage({
     workspaceId: owner.sId,
     agentConfigurationId,
   });
 
+  const usageText = assistantName
+    ? assistantUsageMessage({
+        assistantName,
+        usage: agentUsage.agentUsage,
+        isLoading: agentUsage.isAgentUsageLoading,
+        isError: agentUsage.isAgentUsageError,
+      })
+    : "";
+
   const scopeInfo: Record<
     Exclude<AgentConfigurationScope, "global">,
     {
       label: string;
       color: string;
       icon: typeof UserGroupIcon | typeof PlanetIcon | typeof LockIcon;
       text: string;
+      confirmationModalData: ConfirmationModalDataType;
     }
   > = {
     published: {
       label: "Shared Assistant",
       color: "pink",
       icon: UserGroupIcon,
       text: "Anyone in the workspace can view and edit.",
+      confirmationModalData: {
+        title: "Sharing an assistant",
+        text: "Once shared, the assistant will be visible and editable by members of your workspace.",
+        confirmText: "Share the assistant",
+        variant: "primary",
+      },
     },
     workspace: {
       label: "Company Assistant",
       color: "amber",
       icon: PlanetIcon,
       text: "Activated by default for all members of the workspace.",
+      confirmationModalData: {
+        title: "Moving to Company Assistants",
+        text: "Moving the assistant to Company Assistants will make the assistant editable only by Admins and Builders and add it the “My Assistants” list of all the workspace.",
```

Comment:
Copy seems wrong here.

Reply:
It's the figma copy; but yeah will do a minor tweak

Reply:
I'll do a pass on copy in a new PR.